src/hydrogen/device/ROCm.cpp

```diff
@@ -0,0 +1,163 @@
+#include <El/hydrogen_config.h>
+
+#include <hydrogen/device/GPU.hpp>
+#include <hydrogen/device/gpu/ROCm.hpp>
+
+#include <El/core/MemoryPool.hpp>
+
+namespace hydrogen
+{
+namespace gpu
+{
+namespace
+{
+
+hipStream_t GetNewStream()
+{
+    hipStream_t stream;
+    H_CHECK_HIP(hipStreamCreateWithFlags(&stream, hipStreamNonBlocking));
+    return stream;
+}
+
+hipEvent_t GetNewEvent()
+{
+    hipEvent_t event;
+    H_CHECK_HIP(hipEventCreate(&event));
+    return event;
+}
+
+void FreeStream(hipStream_t stream)
+{
+    if (stream)
+        H_CHECK_HIP(hipStreamDestroy(stream));
+}
+
+void FreeEvent(hipEvent_t event)
+{
+    if (event)
+        H_CHECK_HIP(hipEventDestroy(event));
+}
+
+int ComputeMyDeviceId(unsigned int device_count)
```

Comment:
This is basically duplicated from the CUDA side, but conceptually they do the same thing. Maybe they should be lifted out?

Reply:
Will fix.

Reply:
Fixed, unfortunately triggering a larger refactoring of that stuff.